diff --git a/meson.build b/meson.build
--- a/meson.build
+++ b/meson.build
@@ -55,6 +55,7 @@ endif
 
 with_host_windows = host_machine.system() == 'windows'
 with_host_darwin = host_machine.system() == 'darwin'
+with_host_android = cc.has_define('__ANDROID__')
 
 warnings = [
    '-Werror=enum-int-mismatch',
@@ -295,7 +296,7 @@ have_glx = false
 
 if with_egl
    if cc.has_header('epoxy/egl.h', dependencies: epoxy_dep) and epoxy_dep.get_variable(pkgconfig: 'epoxy_has_egl') == '1'
-      if with_host_windows or with_host_darwin
+      if with_host_windows or with_host_darwin or with_host_android
          have_egl = true
       else
          if libdrm_dep.found()
